pulsar_metrics/metrics/utils.py

```diff
@@ -61,3 +59,24 @@ def psi(new: pd.Series, reference: pd.Series, binned: bool = False):
     psi = ((percents["new"] - percents["ref"]) * np.log(percents["new"] / percents["ref"])).sum()
 
     return psi
+
+
+def mmd(new: pd.DataFrame, reference: pd.DataFrame, kernel="linear", **kwargs):
+    """
+    Calculates the Maximum Mean Discrepency between two samples
+    """
+
+    if isinstance(new, pd.Series):
+        new = new.to_frame()
+
+    if isinstance(reference, pd.Series):
+        reference = reference.to_frame()
+
+    new = new.select_dtypes("number")
+    reference = reference.select_dtypes("number")
+
```

Comment:
Here new and reference are both expected to be generally a pandas dataframe even if with only a single column. However when we single out a single column from a dataframe, it is a pandas series. Here I am just making sure to make a pandas series into a dataframe because the function used later expects a 2D array and not a 1D array.

So to answer you question, if any or neither are a pandas series, they have to be a pandas dataframe as mentioned in the type hint. Let me know if this answers your question and if it is fine to keep it that way.